### PR TITLE
feat(ui): Collapse commit authors in release v2 detail

### PR DESCRIPTION
**Pull request by matejminar**
This PR adds collapsing of commit authors in the release v2 detail page.

image

### CODE REVIEW

File: src/sentry/static/sentry/app/views/releasesV2/detail/overview/commitAuthorBreakdown.tsx [hunk 1]
```diff
@@ -74,14 +86,24 @@ class CommitAuthorBreakdown extends AsyncComponent<Props, State> {
...
     );
 
     if (!sortedAuthorsByNumberOfCommits.length) {
       return null;
     }
 
+    // collapse them
+    const MAX = CommitAuthorBreakdown.MAX_WHEN_COLLAPSED;
+    const initialNumberOfAuthors = sortedAuthorsByNumberOfCommits.length;
+    const canCollapse = initialNumberOfAuthors > MAX;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/releasesV2/detail/overview/commitAuthorBreakdown.tsx]: `canExpand` might be a better name. :bike::house:

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/views/releasesV2/detail/overview/commitAuthorBreakdown.tsx (hunk 1 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/views/releasesV2/detail/overview/commitAuthorBreakdown.tsx]: Thanks!

File: src/sentry/static/sentry/app/views/releasesV2/detail/overview/commitAuthorBreakdown.tsx [hunk 2]
```diff
@@ -46,14 +50,22 @@ class CommitAuthorBreakdown extends AsyncComponent<Props, State> {
   }
 
   getDisplayPercent(authorCommitCount: number): string {
-    const {commitCount} = this.props;
+    const {commits} = this.state;
 
-    const calculatedPercent = round(percent(authorCommitCount, commitCount), 0);
+    const calculatedPercent = round(percent(authorCommitCount, commits.length), 0);
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/releasesV2/detail/overview/commitAuthorBreakdown.tsx]: Why not use `Math.round`?

Comment (matejminar) [reply src/sentry/static/sentry/app/views/releasesV2/detail/overview/commitAuthorBreakdown.tsx]: No reason. I usually use lodash's round when I need decimal places. I was trying how it looked with decimal places and didn't revert back to Math.round afterward... It's changed now.

**Pull request merged**